tianshou/data/batch.py

```diff
@@ -570,10 +782,10 @@ def stack_(self, batches: Sequence[Union[dict, "Batch"]], axis: int = 0) -> None
             # reserved keys
             self.__dict__[key] = Batch()
         for key in keys_partial:
-            for i, batch in enumerate(batches):
+            for i, batch in enumerate(batches):  # type: ignore
```

Comment:
I had ` Incompatible types in assignment (expression has type "Batch", variable has type
"dict[Any, Any] | TBatch") `

The reason is that several isinstance checks before were on Batch instead of on BatchProtocol. I fixed that now and the error disappeared, so the type: ignore is no longer necessary